File: delete_messages.py
import json, requests, sys
print ("Delete all messages from specific channel")
 
username1 = "Ratio"
username2 = "Ratio#1618"
auth_token = "console > application > local > discord > token"
channel_id = "lol"
delete_from_all_users = "False"
 
def get_all_messages(auth, id, last="", prev=[]):
    if not last:
        messages = json.loads(requests.get("http://canary.discordapp.com/api/v6/channels/" + id + "/messages", headers={"authorization": auth}, params={"limit": 100}).content)
    else:
        messages = json.loads(requests.get("http://canary.discordapp.com/api/v6/channels/" + id + "/messages", headers={"authorization": auth}, params={"before" : last, "limit" : 100}).content)
 
    prev = prev + messages
 
    if len(messages) < 100:
        print ("Got to end of channel at " + str(len(prev)) + " messages")
        return prev
    else:
        oldest = sorted(messages, key=lambda x: x["timestamp"], reverse=True)[-1]
        return get_all_messages(auth, id, last=oldest["id"], prev=prev)
 
def delete_all(auth, id, user1, user2, messages):
    print ("Trying to delete all messages in " + id + " from username " + user1)
    for message in messages:
        if (message["author"]["username"] == user1):
            print(message["author"]["username"])
            requests.delete("http://canary.discordapp.com/api/v6/channels/" + id + "/messages/" + message["id"],headers={"authorization": auth})
    print ("All messages were deleted")
 
delete_all(auth_token, channel_id, username1, username2, get_all_messages(auth_token, channel_id))

File: delete_messages.js
function sleep(milliseconds) {
    var start = new Date().getTime();
    while (true) {
        if ((new Date().getTime() - start) > milliseconds) {
            break;
        }
    }
}

function deleteMessages() {

    const authorId = "ENTER_HERE_UR_ID";
    const channelId = "ENTER_HERE_CHANNEL_ID";
    const deleteAfter = "search?author_id=" + authorId + "&min_id=000000000000000000";
    const baseURL = "https://discordapp.com/api/v6/channels/" + channelId + "/messages/";
    const authToken = document.body.appendChild(document.createElement("iframe"))
        .contentWindow.localStorage.token.replace(/"/g, "");

    const headers = {
        "Authorization": authToken
    };

    fetch(baseURL + deleteAfter, {
            headers
        })
        .then(resp => resp.json())
        .then(result => {
            console.log("There are " + result.total_results + " messages left to delete.");
            if (result.total_results == 0) {
                alert("All the messages got deleted. \nPlease refresh the page before pressing 'OK'.");
            }
            result.messages.forEach(function(element) {
                element.forEach(function(message) {
                    if (message.author.id == authorId && message.hit == true) {
                        console.log("Deleting message with Id #" + message.id);
                        console.log(baseURL + message.id);
                        fetch(baseURL + message.id, {
                            headers,
                            method: "DELETE"
                        });
                    }
                });
                sleep(500);
            });
        })
        .then(() => deleteMessages());
}

deleteMessages();
